# Loan approval model
# clearing the data

# step1- importing neccesary libraries

In [1]:
import pandas as pd # for data handling

# step2- reading the data of .csv file

In [2]:
df=pd.read_csv("data/data.csv")

# step3- checking basic terms and analyzing data to fix

In [3]:
df.head()

Loan_ID  Gender Married Dependents Education Self_Employed  \
0  LP001174    Male     Yes          0  Graduate            No   
1  LP002087  Female      No          0  Graduate            No   
2  LP002692    Male     Yes         3+  Graduate           Yes   
3  LP001469    Male      No          0  Graduate           Yes   
4  LP001029    Male      No          0  Graduate            No   

   ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
0             3772              833.0        57.0             360.0   
1             2500                0.0        67.0             360.0   
2             5677             1424.0       100.0             360.0   
3            20166                0.0       650.0             480.0   
4             1853             2840.0       114.0             360.0   

   Credit_History Property_Area Loan_Status  
0             NaN     Semiurban         NaN  
1             1.0         Urban           Y  
2             1.0         Rural           Y  
3             NaN         Urban           Y  
4             1.0         Rural           N

In [38]:
df.isnull().sum()

Married              0
Dependents           0
Education            0
Self_Employed        0
ApplicantIncome      0
CoapplicantIncome    0
LoanAmount           0
Loan_Amount_Term     0
Credit_History       0
Property_Area        0
Loan_Status          0
dtype: int64

Number of null sections in each column

In [5]:
df.describe()

ApplicantIncome  CoapplicantIncome  LoanAmount  Loan_Amount_Term  \
count       981.000000         981.000000  954.000000        961.000000   
mean       5179.795107        1601.916330  142.511530        342.201873   
std        5695.104533        2718.772806   77.421743         65.100602   
min           0.000000           0.000000    9.000000          6.000000   
25%        2875.000000           0.000000  100.000000        360.000000   
50%        3800.000000        1110.000000  126.000000        360.000000   
75%        5516.000000        2365.000000  162.000000        360.000000   
max       81000.000000       41667.000000  700.000000        480.000000   

       Credit_History  
count      902.000000  
mean         0.835920  
std          0.370553  
min          0.000000  
25%          1.000000  
50%          1.000000  
75%          1.000000  
max          1.000000

# step4- removing unwanted column and rows data to train model
Here we create another variable df1 which store the changes.
Also 614 out of 981 is non null rows of loan_status ,so to get accurate or clean data i will also remove these 367 rows.If i dont remove this and filled it with bluff results then almost 34.7 percent data will be bluff.

In [22]:
df=df.drop(columns=['Loan_ID','Gender'])

In [24]:
df=df.dropna(subset=['Loan_Status'])

checking no. of null data of column and working according to them

In [27]:
df.isnull().sum()

Married               3
Dependents           15
Education             0
Self_Employed        32
ApplicantIncome       0
CoapplicantIncome     0
LoanAmount           22
Loan_Amount_Term     14
Credit_History       50
Property_Area         0
Loan_Status           0
dtype: int64

# step5- filling null columns data
We have 6 'object' data type(loan status) and 4 float64 and 1 int64

In [26]:
df # analyzing data for using correct function(mode,mean,median) for each column

Married Dependents     Education Self_Employed  ApplicantIncome  \
1        No          0      Graduate            No             2500   
2       Yes         3+      Graduate           Yes             5677   
3        No          0      Graduate           Yes            20166   
4        No          0      Graduate            No             1853   
5        No          1  Not Graduate           NaN             5191   
..      ...        ...           ...           ...              ...   
973     Yes          1      Graduate            No            12841   
974      No          0  Not Graduate            No             2165   
975     Yes         3+      Graduate            No             6417   
978     Yes          0  Not Graduate           Yes             2609   
980      No          0      Graduate            No             2237   

     CoapplicantIncome  LoanAmount  Loan_Amount_Term  Credit_History  \
1                  0.0        67.0             360.0             1.0   
2               1424.0       100.0             360.0             1.0   
3                  0.0       650.0             480.0             NaN   
4               2840.0       114.0             360.0             1.0   
5                  0.0       132.0             360.0             1.0   
..                 ...         ...               ...             ...   
973            10968.0       349.0             360.0             1.0   
974                0.0        70.0             360.0             1.0   
975                0.0       157.0             180.0             1.0   
978             3449.0       165.0             180.0             0.0   
980                0.0        63.0             480.0             0.0   

    Property_Area Loan_Status  
1           Urban           Y  
2           Rural           Y  
3           Urban           Y  
4           Rural           N  
5       Semiurban           Y  
..            ...         ...  
973     Semiurban           N  
974     Semiurban           Y  
975         Rural           Y  
978         Rural           N  
980     Semiurban           N  

[614 rows x 11 columns]

for: married/self_employed = datatype-object, using-mode, null=yes | LoanAmount/loan_amount_term = datatype-float64, using-median | credit_history = datatype-float64, using-mode | remaining has no null values

In [39]:
mode_column = ['Married','Dependents','Self_Employed','Credit_History']
median_column = ['LoanAmount','Loan_Amount_Term']

In [40]:
for column in df.columns:
    if column in mode_column:
        df[column]=df[column].fillna(df[column].mode()[0])  # we uses [0] to select only one result
    elif column in median_column:
        df[column]=df[column].fillna(df[column].median())


Data after cleaning

In [41]:
df.isnull().sum()

Married              0
Dependents           0
Education            0
Self_Employed        0
ApplicantIncome      0
CoapplicantIncome    0
LoanAmount           0
Loan_Amount_Term     0
Credit_History       0
Property_Area        0
Loan_Status          0
dtype: int64